magento/magento2
# Incorrect order status label in sales update emails sent from the admin for non-default store views issue29263

### Description (*)


Fix issue below:

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#29263: Incorrect order status label in sales update emails sent from the admin for non-default store views 

### Manual testing scenarios (*)

1. In the admin:
    * Go to “Stores > All Stores”, click on “Create Store View”, fill in the required values, set the status to “Enabled” and click on the "Save Store View” button.
    * Go to “Stores > Order Status”, click on the “Pending” status, enter the following data under “Store View Specific Labels”, and click on the “Save Status” button.
        * Default Store View: “Pending (default)”
        * New Store View: “Pending (secondary)”
2. In the frontend:
    * Select the new store view in the store view switcher.
    * Add a product to the cart.
    * Complete the checkout using the “Check / Money order” payment method.
3. In the admin:
    * Go to “Sales > Orders”, and click on the newly created order.
    * On this order view page, enter a comment, click on the “Notify Customer by Email” checkbox and click on the "Submit Comment” button.

**=> An order update email is sent containing “Pending (secondary)” as the order status.**

### Questions or comments


I changed public function getStatusFrontendLabel(?string $code) to public function getStatusFrontendLabel(?string $code, int $storeId = null). We need the $storeId params. Please consider the Semantic Version Checker.

### Contribution checklist (*)
